src/NUnitFramework/tests/Internal/EventListenerTextWriterTests.cs

```diff
@@ -270,64 +280,71 @@ public void TestWriteCharBufferSubstring()
-            Assert.That(TestResult.Outputs.Count, Is.EqualTo(2));
-            Assert.That(TestResult.Outputs[0], Is.EqualTo(expected));
-            Assert.That(TestResult.Outputs[1], Is.EqualTo(expected + NL));
+            Assert.That(ListenerResult.Outputs.Count, Is.EqualTo(2));
+            Assert.That(ListenerResult.Outputs[0], Is.EqualTo(expected));
+            Assert.That(ListenerResult.Outputs[1], Is.EqualTo(expected + NL));
         }
 
         [Test]
         public void TestWriteFloat()
         {
             float value = -5.0f;
 
-            TestWriter.Write(value);
-            TestWriter.WriteLine(value);
+            ListenerWriter.Write(value);
+            ListenerWriter.WriteLine(value);
 
             var expected = "-5";
-            Assert.That(TestResult.Outputs.Count, Is.EqualTo(2));
-            Assert.That(TestResult.Outputs[0], Is.EqualTo(expected));
-            Assert.That(TestResult.Outputs[1], Is.EqualTo(expected + NL));
+            Assert.That(ListenerResult.Outputs.Count, Is.EqualTo(2));
+            Assert.That(ListenerResult.Outputs[0], Is.EqualTo(expected));
+            Assert.That(ListenerResult.Outputs[1], Is.EqualTo(expected + NL));
         }
 
         [Test]
         public void TestWriteLine()
         {
-            TestWriter.WriteLine();
+            ListenerWriter.WriteLine();
 
             var expected = NL;
-            Assert.That(TestResult.Outputs.Count, Is.EqualTo(1));
-            Assert.That(TestResult.Outputs[0], Is.EqualTo(expected));
+            Assert.That(ListenerResult.Outputs.Count, Is.EqualTo(1));
+            Assert.That(ListenerResult.Outputs[0], Is.EqualTo(expected));
         }
 
         #region ITestListener implementation
 
-        private class TestListenerHelper : ITestListener
+        private class TestListenerIntercepter : ITestListener
         {
             public IList<string> Outputs { get; internal set; }
+            public ITestListener DefaultListener { get; internal set; }
 
-            public TestListenerHelper()
+            public TestListenerIntercepter(ITestListener defaultListener)
             {
+                DefaultListener = defaultListener;
                 Outputs = new List<string>();
             }
 
             void ITestListener.TestStarted(ITest test)
             {
-                // Intentionally empty
+                if (DefaultListener != null)
+                    DefaultListener.TestStarted(test);
             }
 
             void ITestListener.TestFinished(ITestResult result)
             {
-                // Intentionally empty
+                if (DefaultListener != null)
+                    DefaultListener.TestFinished(result);
             }
 
             void ITestListener.TestOutput(TestOutput output)
             {
                 Assert.IsNotNull(output);
                 Outputs.Add(output.Text);
+
+                if (DefaultListener != null)
+                    DefaultListener.TestOutput(output);
```

Comment:
No, that would require a new adapter issue to resolve. We definitely don't want to worry about it here because it relates to how the output is used by the runner not with what we produce in the  framework. 

As a thought experiment, imagine that some listener passes messages on to a system that blows up whenever it sees an exclamation point! That would, of course, be a bug in the system in question, just as VS throwing on an empty message seems like a bug to me. In the case where that downstream system can't be fixed the adapter/runner would have to adjust the messages to remove any exclamation points, but other runners would probably still want to see and display them.

You refer to the "original behavior." Do you think this will remain as a problem when this PR is merged?